### ISSUE TITLE
public-api rule reports Unknown segment for @x cross-import paths

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#31]
## Problem

The `public-api` rule with `level: "segments"` incorrectly reports `@x` as an unknown segment for cross-import paths.

**Example:**

```
// File: entities/policies/ui/PoliciesNodesTable.vue
import { useGroups } from "@/entities/groups/@x/policies";
//                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
// ERROR: Unknown segment "@x"
```

The `isUnknownSegment` function calls `extractPotentialSegment`, which splits the path after the layer and takes the second non-group part as a potential segment. For `entities/groups/@x/policies`, it extracts `@x` as the potential segment, which is not in the known segments list.

## Root cause

In `src/rules/public-api/model/is-unknown-segment.ts`, the function `extractPotentialSegment` does not account for `@x` cross-import paths. The path structure `layer/slice/@x/target` is parsed as `layer/slice/segment/...`, treating `@x` as a segment.

Note: `shouldBeFromPublicApi` already handles `@x` correctly via `hasNestedCrossImportPath` and `extractCrossImportInfo`, but `isUnknownSegment` runs before it in `validateAndReport` and catches the error first.

## Expected behavior

The `public-api` rule should detect `@x` cross-import paths and skip the unknown segment check, similar to how other rules handle this.

## Current workaround

```js
featureSliced({
  publicApi: {
    level: "segments",
    ignoreImports: ["**/@x/**"],
  },
})
```

## Related

- #30 (same class of bug in `absolute-relative`, already fixed)

## Environment

- `@conarti/eslint-plugin-feature-sliced`: `2.0.0-rc.11`
- ESLint 9 flat config

**Comment by conarti** [comment]
Fixed in 2.0.0-rc.16 (d388bc9). You can try it with `npm i -D @conarti/eslint-plugin-feature-sliced@next`; it will ship in 2.0.0.
